### PR TITLE
Consider --color option in more places which deal with TerminalWriters

### LINKED ISSUE #856: `--color yes` does not color some output
Specifying `--color yes` on the command line fails to enable colors in some circumstances.

This affects `--fixtures` and `--markers` and possibly other parts of pytest.

This can be worked around by setting the `PY_COLORS` environment variable to `1`.

Reproduction steps:

1. In an empty directory, run: ``py.test --color yes --fixtures | cat``
2. Observe that the names of fixtures are not coloured green

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #856

### CODE REVIEW

**Comment by nicoddemus** [comment]
CC @barneygale

File: _pytest/terminal.py [hunk 1]
```diff
@@ -87,6 +87,7 @@ def __init__(self, code, message, nodeid=None, fslocation=None):
 
 class TerminalReporter:
     def __init__(self, config, file=None):
+        import _pytest.config
```
Comment (RonnyPfannschmidt) [inline _pytest/terminal.py]: why the import?

**Comment by barneygale** [comment]
Why does `_pytest/pastebin.py` not use the new function? Otherwise looks solid to me.

**Comment by nicoddemus** [comment]
Oops, missed that, thanks!

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus after fixing the flake8 issue i think its fine to merge, @hpk42 ?

**Comment by nicoddemus** [comment]
I believe the approach is sound and if we want to refactor it a little (moving it to another module for example) we can do later without much trouble, so I'm merging this.

**Pull request merged**